baybe/acquisition.py

```diff
@@ -1,18 +1,43 @@
 """Adapter for making BoTorch's acquisition functions work with other models."""
+from __future__ import annotations
 
+from functools import partial
 from inspect import signature
-from typing import Any, Callable, List, Optional, Type
+from typing import TYPE_CHECKING, Any, Callable, List, Optional, Type
 
 import gpytorch.distributions
 from attr import define
-from botorch.acquisition import AcquisitionFunction
+from botorch.acquisition import (
+    AcquisitionFunction,
+    ExpectedImprovement,
+    PosteriorMean,
+    ProbabilityOfImprovement,
+    UpperConfidenceBound,
+    qExpectedImprovement,
+    qProbabilityOfImprovement,
+    qUpperConfidenceBound,
+)
 from botorch.models.gpytorch import Model
 from botorch.posteriors import Posterior
 from botorch.posteriors.gpytorch import GPyTorchPosterior
-from torch import Tensor, cat, squeeze
 
 from baybe.surrogates.base import Surrogate
 
+if TYPE_CHECKING:
+    from torch import Tensor
```

Comment:
again the question about whether this even makes sense here? a lot of plain botorch imorts above... wouldnt that mean the lazy torch import is implactless??

Reply:
It's similar to `baybe/recommenders/pure/bayesian/base.py`:

![Screenshot 2024-03-13 at 6 29 11 PM](https://github.com/emdgroup/baybe/assets/5112312/bf5acaec-26ae-4de5-af10-fc83ba563e89)

Reply:
i cant see botorch in that picture despite its import at top of file, can you check?
also, the picture just shows the trace hence the first import of torch, not all traces that would trigger torch import

Reply:
Yes, that's right. I truncated image for readability. `botorch` appears here:
<img width="1202" alt="Screenshot 2024-03-15 at 7 06 24 AM" src="https://github.com/emdgroup/baybe/assets/5112312/8eef49b4-c63e-4efd-b2a7-5315aa748077">